### PR TITLE
Admins Page. Layout Component. Modal Updates. Button Updates

### PR DESCRIPTION
**Pull request by tackc**
#### Please check if the PR fulfills these requirements

- [x] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 

---

#### What kind of change does this PR introduce? (bug fix, feature, docs update, ...)

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

#### What is the current behavior? (You can also link to an open issue here)
Page for viewing all admins did not previously exist

#### What is the new behavior? (if this is a feature change)
- Created page to view all Admins
- Created front end route for `/admins` page
- Created `AdminsService.js` to handle communication with back end
- Created new `Layout` component to wrap pages in and alleviate styling inconsistencies (will need to be implemented on several existing pages once merged)
- Added `edit` and `trash` icons for admins page
- Alphabetized imports and icons on `src/Components/Icon/map.js`
- Added `--DARK_GREEN` color for button hover
- Refactored `Button` component to handle new button variants (`buttonDanger` & `buttonSuccess`)
- Refactored `Modal` component to handle one or two buttons. Add `propTypes` to `Modal`
- Added new `Layout` component wrapper to `ClientsPage` for testing

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
- Although there do not appear to be any breaking changes, there will be documentation updates needed for the "new" Modal, Layout, and Button components. I can either do this as a separate PR since this one is fairly large, or they can be added if you prefer - let me know.

#### Other information

- PLEASE NOTE - THE ADMINS PAGE IS NOT FUNCTIONAL AS THE BACK END ROUTES HAVE NOT BEEN CREATED YET

#### Images (before/ after screenshots, interaction GIFs, ...)

https://github.com/FoodIsLifeBGP/banana-admin/assets/39445764/475b5015-5afa-4120-955b-a05a9458835b

### CODE REVIEW

**Review by sherviniv — APPROVED** [review]
Looks good!

File: src/Components/Modal/index.js [hunk 1]
```diff
@@ -7,9 +7,26 @@ import styles from './style.module.scss';
...
-    modalOpen, setModalOpen, modalContentRef, children, title,
+    modalOpen,
+    setModalOpen,
+    modalContentRef,
+    children,
+    title,
+    buttonsConfig,
   } = props;
 
+  // Default button configuration if none is provided
```
Comment (renburnett) [inline src/Components/Modal/index.js:18]: i think your code here speaks for itself ✅, not sure we need this comment (but we can keep it either way, no biggie)

File: src/Components/Modal/index.js [hunk 2]
```diff
@@ -7,9 +7,26 @@ import styles from './style.module.scss';
...
+  const defaultButtonConfig = [
+    {
+      text: 'Okay',
+      variant: 'buttonPrimary',
+      action: () => setModalOpen(false),
+    },
+  ];
+
+  // Use provided buttonsConfig or default configuration
```
Comment (renburnett) [inline src/Components/Modal/index.js:27]: ditto

File: src/Components/Button/index.js [hunk 3]
```diff
@@ -21,7 +19,12 @@ Button.propTypes = {
   // eslint-disable-next-line react/require-default-props
   style: PropTypes.object,
   action: PropTypes.func.isRequired,
-  variant: PropTypes.string.isRequired,
+  variant: PropTypes.oneOfType([
```
Comment (renburnett) [inline src/Components/Button/index.js:22]: nice use of prop types

File: src/Services/AdminsService.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import ApiService from './ApiService';
+
+const { axiosRequest } = ApiService();
+
+const GetAdmins = async (pageNumber, pageSize) => {
```
Comment (renburnett) [inline src/Services/AdminsService.js:5]: no biggie, but @tackc i think semantically only components and services are `PascalCase`-- i think these methods should be `camelCase`

File: src/Components/Button/index.js (hunk 3 above)
Comment (tackc) [reply src/Components/Button/index.js:22]: Thank you 😊. I realized much of my previous work did not always include them and plan to take a look through to see where they are missing as it helps Storybook with interactivity

File: src/App.js [hunk 5]
```diff
@@ -48,6 +49,14 @@ function App() {
               </AuthWrapper>
             )}
           />
+          <Route
+            path="/admins"
```
Comment (renburnett) [inline src/App.js:53]: hmm for some reason `/admins` is giving me a 404 error
image

File: src/Services/AdminsService.js (hunk 4 above)
Comment (tackc) [reply src/Services/AdminsService.js:5]: If you're ok with it, I will fix this in a separate PR - I tried to keep my casing consistent with the other Services. We should probably update all of them to be semantically correct

File: src/App.js (hunk 5 above)
Comment (tackc) [reply src/App.js:53]: Something seems amiss here...that's not the correct styling for the 404 page. The nav doesn't look right either

File: src/Components/Modal/index.js (hunk 1 above)
Comment (tackc) [reply src/Components/Modal/index.js]: Good call...I removed this comment 😀

File: src/Components/Modal/index.js (hunk 2 above)
Comment (tackc) [reply src/Components/Modal/index.js]: Removed!

File: src/App.js (hunk 5 above)
Comment (renburnett) [reply src/App.js:53]: @tackc oh crap i'm a fool, i checked out the wrong branch! hang on sorry

**Review by renburnett — APPROVED** [review]
looks awesome! love the styling! sorry for the confusion earlier.

File: src/Services/AdminsService.js (hunk 4 above)
Comment (renburnett) [reply src/Services/AdminsService.js:5]: no prob! its small potatoes. feel free to leave it

File: src/App.js (hunk 5 above)
Comment (renburnett) [reply src/App.js:53]: works as described sorry, i had a long day

**Pull request merged**